### ISSUE TITLE
Pacman reads Terabytes when processing package changes 

### ISSUE THREAD

**Issue by xeyownt** [msys2/MSYS2-packages#1788]
When running `pacman -Syuu` for the second time, pacman goes unexpectedly slow during the phase processing package changes. When I say "unexpectedly", I mean it does not seem to match the speed for that kind of operation compared to other similar operation on the same PC.

As an example, pacman is currently processing 140 packages, and in roughly 10 minutes, it only processed 28 packages. Checking the taskmanager, `pacman.exe` is reported to have **900GB** for *I/O Read Bytes*, and to have regularly **1.3GB** for *I/O Delta Read Bytes*. This seems quite a lot to say the least. It will probably complete with more than **3TB** at completion.

Any idea what could be causing this excessive read / unexpected slow down?

For completeness, this is running on a "exotic" platform:
* Windows 10,
* With MacAfee anti-virus.
* Running **inside VirtualBox 6.0.14**, with 8 GB RAM allocated.
* Host is Debian-based with SSD >500MB/s read speed.

This is maybe an artefact of running within VirtualBox but I can't see why this would be the case.

**Comment by StarWolf3000** [comment]
This is by no means exotic, it is very common to virtualize Windows in a hypervisor, independent of the host operating system.

`pacman` does a lot of I/O when upgrading packages. Is the virtual hard disk defined as an SSD drive in the virtual machine? Windows will optimize I/O, when this is the case. Also your McAfee might slow down the upgrade process and do unnecessary I/O, if the MSYS2 installation path is not added to an exclusion list.

1.3 GB for Delta Read Bytes sounds normal, but 900 GB for I/O Read Bytes is a bit too much. Is this just for this running process at this point or summed up over time?

**Comment by xeyownt** [comment]
Indeed, the virtual disk is detected as solid-state drive (seen in win disk optimizer).

I cannot unfortunately change the McAfee configuration. I'm administrator, but the AV is managed by corporate IT, and all configuration changes is an up-hill battle ^^. Note however that MSYS2 is otherwise surprisingly fast, even in the virtual environment. I do run clang compilation at native-like speed for instance.

So the 900GB (and indeed it topped around the 3TB at the end) was solely for the time of the "pacman -Syuu" process. I just opened a new terminal, started the upgrade, and then the I/O Read Bytes started skyrocketing.

As said, MSYS2 is surprisingly fast in this setup. I am only surprised by the much slower speed of pacman, and high I/O usage. This is not a real problem for me. So if there is no obvious reason or suggestion, I'm happy to close this ticket.
